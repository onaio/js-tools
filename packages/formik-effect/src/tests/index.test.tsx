import { mount } from 'enzyme';
import flushPromises from 'flush-promises';
import { Field, Form, Formik } from 'formik';
import React from 'react';
import FormikEffect from '..';

describe('formik-effect', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('works as expected', async () => {
    const mock = jest.fn();

    const SignupForm = () => (
      <div>
        <h1>My Cool Form with a SideEffect</h1>
        {/* tslint:disable jsx-no-lambda */}
        <Formik onSubmit={jest.fn()} initialValues={{ firstName: '', lastName: '', email: '' }}>
          {() => (
            <Form translate="yes">
              <FormikEffect onChange={mock} />
              <Field id="firstName" name="firstName" placeholder="First Name" />
              <Field id="lastName" name="lastName" placeholder="Last Name" />
              <Field name="email" type="email" placeholder="Email Address" />
              <button type="submit">Submit</button>
            </Form>
          )}
        </Formik>
        {/* tslint:enable jsx-no-lambda */}
      </div>
    );

    const wrapper = mount(<SignupForm />);

    wrapper
      .find('Form Field#firstName')
      .simulate('change', { target: { value: 'John', name: 'firstName' } });
    wrapper
      .find('Form #lastName')
      .last()
      .simulate('change', { target: { value: 'Doe', name: 'lastName' } });

    await flushPromises();
    wrapper.update();

    expect(mock).toHaveBeenCalledTimes(5);

    expect(mock.mock.calls[2]).toMatchSnapshot();
  });
});
