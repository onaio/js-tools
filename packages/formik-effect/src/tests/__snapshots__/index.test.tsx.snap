// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formik-effect works as expected 1`] = `
Array [
  Object {
    "dirty": true,
    "errors": Object {},
    "getFieldHelpers": [Function],
    "getFieldMeta": [Function],
    "getFieldProps": [Function],
    "handleBlur": [Function],
    "handleChange": [Function],
    "handleReset": [Function],
    "handleSubmit": [Function],
    "initialErrors": Object {},
    "initialStatus": undefined,
    "initialTouched": Object {},
    "initialValues": Object {
      "email": "",
      "firstName": "",
      "lastName": "",
    },
    "isSubmitting": false,
    "isValid": true,
    "isValidating": true,
    "registerField": [Function],
    "resetForm": [Function],
    "setErrors": [Function],
    "setFieldError": [Function],
    "setFieldTouched": [Function],
    "setFieldValue": [Function],
    "setFormikState": [Function],
    "setStatus": [Function],
    "setSubmitting": [Function],
    "setTouched": [Function],
    "setValues": [Function],
    "status": undefined,
    "submitCount": 0,
    "submitForm": [Function],
    "touched": Object {},
    "unregisterField": [Function],
    "validateField": [Function],
    "validateForm": [Function],
    "validateOnBlur": true,
    "validateOnChange": true,
    "validateOnMount": false,
    "values": Object {
      "email": "",
      "firstName": "John",
      "lastName": "",
    },
  },
  Object {
    "dirty": true,
    "errors": Object {},
    "getFieldHelpers": [Function],
    "getFieldMeta": [Function],
    "getFieldProps": [Function],
    "handleBlur": [Function],
    "handleChange": [Function],
    "handleReset": [Function],
    "handleSubmit": [Function],
    "initialErrors": Object {},
    "initialStatus": undefined,
    "initialTouched": Object {},
    "initialValues": Object {
      "email": "",
      "firstName": "",
      "lastName": "",
    },
    "isSubmitting": false,
    "isValid": true,
    "isValidating": true,
    "registerField": [Function],
    "resetForm": [Function],
    "setErrors": [Function],
    "setFieldError": [Function],
    "setFieldTouched": [Function],
    "setFieldValue": [Function],
    "setFormikState": [Function],
    "setStatus": [Function],
    "setSubmitting": [Function],
    "setTouched": [Function],
    "setValues": [Function],
    "status": undefined,
    "submitCount": 0,
    "submitForm": [Function],
    "touched": Object {},
    "unregisterField": [Function],
    "validateField": [Function],
    "validateForm": [Function],
    "validateOnBlur": true,
    "validateOnChange": true,
    "validateOnMount": false,
    "values": Object {
      "email": "",
      "firstName": "John",
      "lastName": "Doe",
    },
  },
]
`;
